### ISSUE TITLE
selector query "*|E" not work when tag name contains upper characters and lower characters

### ISSUE THREAD

**Issue by ckfun0skf** [jhy/jsoup#1208]
I have an XML document with tag `<ns1:MyXmlTag>` and `<ns2:MyXmlTag>`, I am trying to extract both with `doc.select("*|MyXmlTag")` , but it return nothing.

After running some test in debugger, I find that "*|MyXmlTag" was parsed to ":myxmltag" and return false while evaluating in `TagEndsWith.matches()`.

**Comment by krystiangorecki** [comment]
I can reproduce your this behavior in latest version 1.11.3 but I found something even more bizzare. 
Try running this code with and without line `System.out.println(doc);` When running with this line commented out the output is empty, but when this line executes the output is as expected.
```
public static void main(String[] args) {
    Document doc = Jsoup.parse("<ns1:MyXmlTag>1111</ns1:MyXmlTag><ns2:MyXmlTag>2222</ns2:MyXmlTag>", "", Parser.xmlParser());
    System.out.println(doc); // run the code with and without this line to get different output :)
    System.out.println(doc.select("*|MyXmlTag"));
}

**Comment by mulinJiao** [comment]
In jsoup, using `Parser.xmlParser()` to generate the DOM is not the same as using `Parser.htmlParser()`.
E.g.
tagName in `Parser.xmlParser()` - case sensitive 
tagName in `Parser.htmlParser()` - lower case.

And when we use the selector - `*|MyXmlTag` - the string is converted to lowercase.
Therefore, in the case `Parser.xmlParser()`, the selector will find nothing.

**Comment by dota17** [comment]
In jsoup, using `Parser.xmlParser()` to generate the DOM is not the same as using `Parser.htmlParser()`.
E.g.
tagName in `Parser.xmlParser()` - case sensitive 
tagName in `Parser.htmlParser()` - lower case.

And when we use the selector - `*|MyXmlTag` - the string is converted to lowercase.
Therefore, in the case `Parser.xmlParser()`, the selector will find nothing.

**Comment by jhy** [comment]
Thanks, fixed in #1257
